src/components/HistoryTables/MsigProposals/Proposal.tsx

```diff
@@ -39,111 +35,96 @@ export default function ProposalDetail({
   cancel
 }: ProposalDetailProps) {
   const router = useRouter()
-  const { coinType, networkName } = useEnvironment()
   const logger = useLogger()
-  let {
+  const { coinType } = useEnvironment()
+
+  const [seeMore, setSeeMore] = useState(false)
+
+  // Ensure network cointype for address
+  const address = useMemo<string>(
+    () => convertAddrToPrefix(msigAddress, coinType),
+    [msigAddress, coinType]
+  )
+
+  // Get safe proposals
+  const {
     data: msigTxsData,
     loading: msigTxsLoading,
-    error: _msigTxsError
+    error: msigTxsError
   } = useMsigPendingQuery({
-    variables: {
-      address: convertAddrToPrefix(msigAddress, coinType)
-    },
+    variables: { address },
     pollInterval: 0
   })
 
-  const proposal = useMemo(() => {
-    if (!msigTxsLoading && !_msigTxsError) {
-      return msigTxsData?.msigPending.find(p => p.id === id)
-    }
-    return null
-  }, [msigTxsData, msigTxsLoading, _msigTxsError, id])
-
-  const [seeMore, setSeeMore] = useState(false)
+  // Find current proposal
+  const proposal = useMemo(
+    () => msigTxsData?.msigPending.find(p => p.id === id) || null,
+    [msigTxsData, id]
+  )
 
+  // Load the actor state
   const {
     data: actorData,
     loading: actorLoading,
+    notFound: actorNotFound,
     error: actorError
-  } = useActorQuery({
-    variables: { address: convertAddrToPrefix(msigAddress, coinType) }
-  })
+  } = useStateReadState<MsigState>(address)
+
+  // Get the actor name from the actor code
+  const actorName = useMemo<string | null>(
+    () => (actorData ? getActorName(actorData.Code['/']) : null),
+    [actorData]
+  )
 
-  const loadMsigState = useMemo<boolean>(() => {
-    try {
-      return (
-        !!actorData &&
-        decodeActorCID(actorData.actor.Code, networkName).includes('multisig')
-      )
-    } catch (e) {
-      logger.error(e)
-      return false
-    }
-  }, [actorData, networkName, logger])
+  // Interpret state data as msig state if valid
+  const msigState = useMemo<MsigState | null>(() => {
+    const isMsigActor = actorName && actorName.includes('multisig')
+    return isMsigActor && actorData ? (actorData.State as MsigState) : null
+  }, [actorData, actorName])
 
-  const {
-    data: stateData,
-    loading: stateLoading,
-    error: stateError
-  } = useStateReadStateQuery<MsigState>({
-    variables: { address: convertAddrToPrefix(msigAddress, coinType) },
-    skip: !loadMsigState
```

Comment:
Skip was not being handled by `useStateReadStateQuery`, see comment in `useStateReadState.ts`